test/test_utility_functions.py

```diff
@@ -151,3 +152,17 @@ def test_does_not_increment_without_stream_send(self):
 
         third_stream_id = c.get_next_available_stream_id()
         assert third_stream_id == (first_stream_id + 2)
+
+
+class TestExtractHeader(object):
+
+    example_request_headers = [
+        (':authority', 'example.com'),
+        (':path', '/'),
+        (':scheme', 'https'),
+        (':method', 'GET'),
+    ]
+
+    def test_extract_header_method(self):
+        assert extract_method_header(self.example_request_headers) \
+            == u'GET'
```

Comment:
I've made the suggested changes but not included the additional core tests. I'll do that now as I have work to do tomorrow.